src/backend/base/langflow/components/vectorstores/OpenSearch.py

```diff
@@ -0,0 +1,292 @@
+            self.status = ""
+            return
+
+        documents = []
+        for _input in self.ingest_data or []:
+            if isinstance(_input, Data):
+                documents.append(_input.to_lc_document())
+            else:
+                error_message = "Vector Store Inputs must be Data objects."
+                logger.error(error_message)
+                raise ValueError(error_message)
+
+        if documents and self.embedding is not None:
+            logger.debug(f"Adding {len(documents)} documents to the Vector Store.")
+            try:
+                vector_store.add_documents(documents)
+            except Exception as e:
+                error_message = f"Error adding documents to Vector Store: {str(e)}"
+                logger.error(error_message)
+                logger.error(f"Traceback: {traceback.format_exc()}")
+                raise Exception(error_message)
+        else:
+            logger.debug("No documents to add to the Vector Store.")
+
+    def search(self, query: Optional[str] = None) -> List[Dict[str, Any]]:
+        """
+        Search for similar documents in the vector store or retrieve all documents if no query is provided.
+        """
+        try:
+            vector_store = self.build_vector_store()
+            search_kwargs = {"k": self.number_of_results, "score_threshold": self.search_score_threshold}
+
+            if query:
+                search_type = self.search_type.lower()
+                if search_type == "similarity":
+                    results = vector_store.similarity_search_with_score(query, **search_kwargs)
+                    # Handle results with scores
+                    return [
+                        {"page_content": doc.page_content, "metadata": doc.metadata, "score": score}
+                        for doc, score in results
+                    ]
+                elif search_type == "mmr":
+                    results = vector_store.max_marginal_relevance_search(query, **search_kwargs)
+                    # Handle results without scores
+                    return [{"page_content": doc.page_content, "metadata": doc.metadata} for doc in results]
+                else:
+                    raise ValueError(f"Invalid search type: {self.search_type}")
+            else:
+                # Retrieve all documents when no query is provided
+                results = self.get_all_documents(vector_store, **search_kwargs)
+                return [
+                    {"page_content": doc.page_content, "metadata": doc.metadata, "score": score}
+                    for doc, score in results
+                ]
+        except Exception as e:
+            error_message = f"Error during search: {str(e)}"
+            logger.error(error_message)
+            logger.error(f"Traceback: {traceback.format_exc()}")
+            raise Exception(error_message)
+
+    def get_all_documents(self, vector_store: OpenSearchVectorSearch, **kwargs) -> List[Tuple[Document, float]]:
```

Comment:
Unless this behavior makes more sense than the default, and we have a specific use case around this, I would lean towards using the default behavior in langchain's search (which may be to retrieve all anyway?)

Reply:
Yes, you are correct, it is really the default behavior. I fixed this as well on my last push.